pennylane/templates/subroutines/qrom.py

```diff
@@ -0,0 +1,300 @@
+        returning ``work_wires`` to their initial state. This technique is able to work with ``work_wires`` that are not
+        initialized to zero.
+
+    """
+
+    def __init__(
+        self, bitstrings, control_wires, target_wires, work_wires, clean=True, id=None
+    ):  # pylint: disable=too-many-arguments
+
+        control_wires = qml.wires.Wires(control_wires)
+        target_wires = qml.wires.Wires(target_wires)
+
+        if work_wires:
+            work_wires = qml.wires.Wires(work_wires)
+
+        self.hyperparameters["bitstrings"] = bitstrings
+        self.hyperparameters["control_wires"] = control_wires
+        self.hyperparameters["target_wires"] = target_wires
+        self.hyperparameters["work_wires"] = work_wires
+        self.hyperparameters["clean"] = clean
+
+        if work_wires:
+            if any(wire in work_wires for wire in control_wires):
+                raise ValueError("Control wires should be different from work wires.")
+
+            if any(wire in work_wires for wire in target_wires):
+                raise ValueError("Target wires should be different from work wires.")
+
+        if any(wire in control_wires for wire in target_wires):
+            raise ValueError("Target wires should be different from control wires.")
+
+        all_wires = target_wires + control_wires + work_wires
+        super().__init__(wires=all_wires, id=id)
+
+    def _flatten(self):
+        data = (self.hyperparameters["bitstrings"],)
+        metadata = tuple(
+            (key, value) for key, value in self.hyperparameters.items() if key != "bitstrings"
+        )
+        return data, metadata
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        bitstrings = data[0]
+        hyperparams_dict = dict(metadata)
+        return cls(bitstrings, **hyperparams_dict)
+
+    def __repr__(self):
+        return f"QROM(control_wires={self.control_wires}, target_wires={self.target_wires},  work_wires={self.work_wires})"
+
+    def map_wires(self, wire_map: dict):
+        new_target_wires = [
+            wire_map.get(wire, wire) for wire in self.hyperparameters["target_wires"]
+        ]
+        new_control_wires = [
+            wire_map.get(wire, wire) for wire in self.hyperparameters["control_wires"]
+        ]
+
+        new_work_wires = []
+        if self.hyperparameters["work_wires"]:
+            new_work_wires = [
+                wire_map.get(wire, wire) for wire in self.hyperparameters["work_wires"]
+            ]
+
+        return QROM(
+            self.bitstrings, new_control_wires, new_target_wires, new_work_wires, self.clean
+        )
+
+    def __copy__(self):
+        """Copy this op"""
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+
+        new_data = copy.copy(self.data)
+
+        for attr, value in vars(self).items():
+            if attr != "data":
+                setattr(copied_op, attr, value)
+
+        copied_op.data = new_data
```

Comment:
QROM has no data, everything is stored as hyperparameters! I am not sure why copy has been implemented this way. I doubt anyone will be subclassing from QROM and adding to data so its probably safe to remove the following lines, but let me know if any tests fail: 

```suggestion
        for attr, value in vars(self).items():
                setattr(copied_op, attr, value)
```